### DISCUSSION TITLE
CEST Obrigatório para modalidade 500

### DISCUSSION THREAD

**Discussion by gustavosg2** [TadaSoftware/PyNFe#287] [Q&A]
A lib não permite setar o CEST quando adiciona produto, mas para modalidade de ICSM=500 o CEST é obrigatório, então não consigo emitir uma nova válida por isso. Alguem consegue dar uma visão do assunto?

**Comment by gustavosg2** [comment]
Vi que o código que serializa o CEST ta comentado e queria entender melhor o pq disso

https://github.com/TadaSoftware/PyNFe/blob/67b5bca71c5d093fdfb34879db9da5e3901d26d6/pynfe/processamento/serializacao.py#L302C11-L302C16

/PyNFe/pynfe/processamento/serializacao.py

![image](https://github.com/TadaSoftware/PyNFe/assets/28692182/f9ae5e70-958a-4cd2-a36a-9ba9de28614a)

**Comment by gustavosg2** [comment]
Resolvi o caso enxertando o CEST no XML logo depois do NCM.
Deixar o código aqui para quem tiver com o mesmo problema que eu, só vc adicionar entre a serialização da nfe e a assinatura (logico que vc vai precisar adaptar pro CEST não ficar fixo):

```
# serialização
serializador = SerializacaoXML(_fonte_dados, homologacao=homologacao)
nfe = serializador.exportar()

# ENXERTANTO O CEST NA MARRA LOGO DEPOIS DO NCM DE CADA ITEM:
from lxml import etree
# Levanta os elementos NCM
elements_ncm = nfe.xpath(".//NCM")

# Para cada elemento NCM, crie um novo elemento CEST e insira após o NCM
for element_ncm in elements_ncm:
    element_cest = etree.Element("CEST")
    element_cest.text = '1003600'
    
    parent = element_ncm.getparent()
    index_ncm = parent.index(element_ncm)
    parent.insert(index_ncm + 1, element_cest)

# assinatura
a1 = AssinaturaA1(certificado, senha)
xml = a1.assinar(nfe)

# envio
con = ComunicacaoSefaz(uf, certificado, senha, homologacao)
envio = con.autorizacao(modelo='nfe', nota_fiscal=xml)

```

**Comment by FilipeRafaeli** [comment]
só pra reativar a discussão, alguem tem motivo pra estar comentado? ou da pra descomentar e commitar?
